lib/internal/util/inspect.js

```diff
@@ -780,7 +780,7 @@ function getPrefix(constructor, tag, fallback, size = '') {
     return `[${fallback}${size}: null prototype] `;
   }
 
-  if (tag !== '' && constructor !== tag) {
+  if (tag !== '' && !constructor.includes(tag)) {
     return `${constructor}${size} [${tag}] `;
   }
   return `${constructor}${size} `;
```

Comment:
```suggestion
  let result = `${constructor}${size} `;
  if (tag !== '') {
    const position =  constructor.indexOf(tag);
    if (position === -1) {
      result += `[${tag}] `;
    } else {
      const endPos = position + tag.length;
      if (endPos !== constructor.length &&
        constructor[endPos + 1] === constructor[endPos + 1].toLowerCase()) {
        result += `[${tag}] `;
      }
    }
  }
  return result;
```

I believe that would make the check a tad 'safer' in a way that it would detect lowercase characters next to the tag. That is a rare case and I think this is therefore fine to do.